docs/application/native/guides/machine-learning/machine-learning-training.md

```diff
@@ -0,0 +1,469 @@
+
+Type | Key | value | Description
+---------- | --- | ----- | -----------
+(Universal props)                   |                    |               | Universal props that applies to every layer
+&#xfeff;                            | learning_rate      | (float)       | Initial learning rate for the optimizer
+`ML_TRAIN_OPTIMIZER_TYPE_SGD`       |                    |               | Stochastic Gradient Descent optimizer
+`ML_TRAIN_OPTIMIZER_TYPE_ADAM`      |                    |               | Adam optimizer
+&#xfeff;                            | decay_steps        | (float)       | Decay steps
+&#xfeff;                            | decay_rate         | (float)       | Decay rate
+&#xfeff;                            | beta1              | (float)       | beta1 coefficient for adam
+&#xfeff;                            | beta2              | (float)       | beta2 coefficient for adam
+&#xfeff;                            | epsilon            | (float)       | epsilon coefficient for adam
+
+### Dataset
+
+Dataset is in charge of feeding data into the model.
+The dataset can either be created from a callback function or created from a file.
+For more information, see [configure the model](#configure-the-model) section.
+
+Following code is example of handling dataset:
+
+```c
+// Create dataset
+ml_train_dataset_h dataset;
+ml_train_dataset_create_with_generator(&dataset, generator_train_cb, generator_valid_cb, generator_test_cb);
+
+// configure dataset
+ml_train_dataset_set_property(dataset, "buffer_size=100", NULL);
+
+// after setting a dataset to model,
+// you do not need to destroy dataset since ownership is transferred to the model.
+ml_train_model_set_dataset(model, dataset);
+```
+
+## Construct a model
+
+A model can be constructed with `ml_train_model_construct`.
+If you have a file that describes the model, the file can be used to construct initially with `ml_train_model_construct_with_file`.
+Even if the model is constructed from a file, switching, modifying, or setting a component is possible only until you compile with `ml_train_model_compile`.
+
+### Construct a model from a description file
+
+As of now, only INI formatted files `*.ini` is supported to construct a model from a file.
+
+#### Create a model from INI formatted file
+
+Special sections `[Model]` and `[Dataset]` are respectively referring to `model` and `dataset` object.
+Rest of INI sections map to a `layer`. Keys and values from each section set properties of the layer.
+All keys and values are treated as case-insensitive.
+
+For example:
+
+```ini
+[Model] # Special section that describes model itself
+Type = NeuralNetwork  # Model Type : only NeuralNetwork is supported as of now
+Optimizer = adam  # Optimizer : sgd (stochastic gradient decent),
+          #             adam (Adaptive Moment Estimation)
+#### optimizer related properties
+Learning_rate = 0.0001  # Learning Rate for the optimizer
+Decay_rate = 0.96 # for the decay_rate for the decayed learning rate
```

Comment:
Since it is not actually decayed learning rate, changed as below

```suggestion
Decay_rate = 0.96 # The decay rate for decaying the learning rate
```